### PR TITLE
MHV-53632 fixed missing facilities in cerner warning

### PR DESCRIPTION
**Pull request by KolbyVA**
## Summary
Fixes missing facilities in the warning banner for cerner facilities. This is done by no longer using a paginated api call, but instead using information from the redux store.

## Related issue(s)
https://jira.devops.va.gov/browse/MHV-53632
> User association with Cerner facility do not see the banner notifying them that they have association with Cerner facilities and need to navigate to VA Health to access their records. This scenario only happens when a user is assigned to a facility that is not coming back from a Facility API response. 
> 
> GIVEN An authenticated user has associations with Cerner Facilities
> 
> WHEN A user lands on SM pages
> 
> AND Application sends a request to Facility API https://api.va.gov/v1/facilities/va?ids=vha_989,vha_123 to retrieve information about that facility to display a full facility name in a banner
> 
> THEN Certain Facilities are not included in the response. Due to code logic, the banner is not displayed
> 
> 
> 
> 
> We need to research to understand why Facility API is not including some facilities in a response
> Need to handle the scenario above in our code. Although the information about facility was not received from Facility API, we still need to inform a user about their association with Cerner and include banner with navigation button to VA Health
> need specific content for this scenario 
> restructure the logic in SM app to handle this 
> 
> Feature Flag Y/N ? N
> 
> Manual Testing Y/N ? Y
> 
> Automated Testing Y/N ? Y
> 
> Accessibility Testing Y/N ? N
> 
> UCD Validation Y/N N

## Testing done
Unit tests updated and passing. Coverage for warning banner is above 90%.

## Screenshots
No UI change.

## What areas of the site does it impact?
my-health/secure-messaging

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by KyleCardwell — DISMISSED** [review]
LGTM

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
Great work. I see some merge conflicts need to be resolved. One thing I noticed while testing in RI which is not the typical case, but something we need to handle. that drupal static data on facilities comes from `vamc-ehr.json` response. If that response fails, entire SM gets "paralyzed". We should be able to add a condition to overcome this scenario where this drupalStaticData redux store is not containing any value

File: src/applications/mhv/secure-messaging/components/MessageList/FolderHeader.jsx [hunk 1]
```diff
@@ -44,17 +43,20 @@ const FolderHeader = props => {
...
+        facility => facility.isCerner,
+      );
       if (cernerTransition556T30) {
-        const facilities = cernerFacilities.filter(
-          facility => facility.isCerner && facility.facilityId !== '556',
+        cernerFacilitiesFiltered = cernerFacilitiesFiltered.filter(
+          facility => facility.facilityId !== '556',
         );
-        return facilities.length > 0;
+        return cernerFacilitiesFiltered;
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/MessageList/FolderHeader.jsx:55]: I think this line may be redundant now that both returns are the same

**Review by alex-morgun-1 — DISMISSED** [review]

**Review by rmessina1010 — DISMISSED** [review]
LGTM

**Review by alex-morgun-1 — APPROVED** [review]

**Pull request merged**